Drivers/CAN/can.c

```diff
@@ -0,0 +1,71 @@
+/*
+    CAN DRIVER Source File
+    Link to part Datasheet for reference:
+    https://www.st.com/resource/en/user_manual/um1725-description-of-stm32f4-hal-and-lowlayer-drivers-stmicroelectronics.pdf
+    
+    Author: Hamza Iqbal
+*/
+
+#include "can.h"
+
+
+void CAN_Handler_Init(CAN_HandleTypeDef* CAN_Line)
+{
+    return;
+}
+
+
+void start_CAN()
+{
+    return;
+}
+
+CAN_msg_t serializeCANMsg(uint32_t id, uint8_t len, uint8_t* data, uint8_t bus, CAN_HandleTypeDef can_line)
+{
+    return;
+}
+
+int sendMessageCAN1(uint32_t id, uint8_t len, const uint8_t *data)
+{
+    return;
+}
+
+int sendMessageCAN2(uint32_t id, uint8_t len, const uint8_t *data)
+{
+    return;
+}
+
+int sendMessageCAN3(uint32_t id, uint8_t len, const uint8_t *data)
+{
+    return;
+}
+
+void incoming_callback_CAN(const CAN_msg_t &msg)
+{
+    return;
+}
+
+
+
+
+__attribute__((weak)) void CAN_BMS_acc_status             (const CAN_msg_t &msg){return;}
```

Comment:
For this iteration of the driver, i don't think that we should add the callback declarations in here, even if they are weak declarations.